src/uri.rs

```diff
@@ -0,0 +1,261 @@
+#![allow(dead_code)]
```

Comment:
I don't think we should allow the merging of feature branches with this line.  There appears to be a lot of dead code in this review and seeing the build output would help a lot.

Reply:
Yeah that sounds good to me. 
Though without this or non module dead code many of the types that are used for the individual EndPoint enums thing, all get marked as dead code, which i'm still a bit confused by, because they do not get constructed. But they need to exist for the enum to know what their types are? So thats kinda confusing to me.